Ponca/src/Fitting/basket.h

```diff
@@ -76,11 +142,6 @@ namespace internal
     }                                                                                                 \
     WRITE_BASKET_SINGLE_HOST_FUNCTIONS
 
-#define BASKETDIFF_TP(I) template <class, class, int, typename> class Ext##I = internal::ForwardDiff
-#define BSKW typename BasketType::WeightFunction
-#define BSKP typename BasketType::DataPoint
-
-
     /*!
          \brief Aggregator class used to declare specialized structures with derivatives computations, using CRTP
```

Comment:
I believe that CRTP is not really used: no class derives from another one using itself as a template argument like `struct Derived : Base<Derived> {};`, right?

Reply:
CRTP isn't used directly for the basket type but it is used to combine all of the extension types into a single aggregate type, with `BasketAggregate`, `BasketDiffAggregate`, etc. (e.g. in `BasketAggregateImpl`, the `Aggregate` template parameter gets passed down to each extension type recursively).